pkg/addr/addr.go

```diff
@@ -14,9 +16,25 @@ import (
 const DefaultWeight = 1
 
 // PublicAddressToString formats a Public API TCPAddress as a string.
+//
+// If Ipv6, the bytes should be ordered big-endian. When formatted as a
+// string, the IP address should be enclosed in square brackets followed by
+// the port.
 func PublicAddressToString(addr *public.TcpAddress) string {
```

Comment:
This function has been updated to format both Ipv4 and Ipv6 IP addresses. When Ipv6, the IP address is enclosed in square brackets to separate it from the port.

I changed the event destination to be IPv6 and updated the tests accordingly.

I could separate the IPv4/IPv6 formatting changes from this PR, but the changes were related to handling IPv6 possibility in `TapEvent`s and are confined to these tests only.